### PR TITLE
Customer service

### PR DESCRIPTION
**Pull request by Nikita-Exeption**

### CODE REVIEW

File: src/main/java/org/nikita/chiken_bell/core/exception/PhoneExistException.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package org.nikita.chiken_bell.core.exception;
+
+public class PhoneExistException extends RuntimeException {
```
Comment (zxRasp) [inline src/main/java/org/nikita/chiken_bell/core/exception/PhoneExistException.java]: Change name to "PhoneAlreadyExistException"

File: src/main/java/org/nikita/chiken_bell/core/service/CustomerService.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+public interface CustomerService {
+
+    Customer create(String name, String phone, String address);
+
+    Customer update(String id, String name, String phone, String address);
+
+    Optional<Customer> getById(String id);
+
+    Optional<Customer> getByPhone(String phone);
```
Comment (zxRasp) [inline src/main/java/org/nikita/chiken_bell/core/service/CustomerService.java]: change name to "findByPhone"

File: src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.*;
+
+public class CustomerServiceImpl implements CustomerService {
+
+    private final Set<Customer> customers = new HashSet<>();
+
+
+    @Override
+    public Customer create(String name, String phone, String address) {
+        if (customers.stream().filter(x -> x.getPhone().equals(phone)).findFirst().isPresent()){
```
Comment (zxRasp) [inline src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java]: use "findByPhone" method.

File: src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        Customer customer = new Customer(name, phone, address);
+        customers.add(customer);
+        return customer;
+    }
+
+    @Override
+    public Customer update(String id, String name, String phone, String address) {
+        Customer customer =getById(id).orElseThrow(NotFoundCustomerException::new);
+        customer.setName(name);
+        customer.setPhone(phone);
```
Comment (zxRasp) [inline src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java:36]: 1. we need check, that phone is not null
2.  we need check, that phone is unique

File: src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @Override
+    public Collection<Customer> getAll() {
+        return Collections.unmodifiableSet(customers);
+    }
+
+    @Override
+    public void deleteById(String id) {
+        customers.remove(getById(id).orElseThrow(NotFoundCustomerException::new));
```
Comment (zxRasp) [inline src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java]: Please rename exception to "CustomerNotFoundException"

**Review by zxRasp — CHANGES_REQUESTED** [review]

File: src/main/java/org/nikita/chiken_bell/core/service/CustomerService.java (hunk 2 above)
Comment (Nikita-Exeption) [reply src/main/java/org/nikita/chiken_bell/core/service/CustomerService.java]: Ok

File: src/main/java/org/nikita/chiken_bell/core/exception/PhoneExistException.java (hunk 1 above)
Comment (Nikita-Exeption) [reply src/main/java/org/nikita/chiken_bell/core/exception/PhoneExistException.java]: Ok

File: src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java (hunk 3 above)
Comment (Nikita-Exeption) [reply src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java]: ok

File: src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java (hunk 4 above)
Comment (Nikita-Exeption) [reply src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java:36]: What we need do , if phone null ?

File: src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java (hunk 5 above)
Comment (Nikita-Exeption) [reply src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java]: Ok

**Review by zxRasp — APPROVED** [review]

**Pull request merged**